apollo-backend/src/main/java/io/logz/apollo/kubernetes/KubernetesMonitor.java

```diff
@@ -146,6 +151,20 @@ public void monitor() {
         }
     }
 
+    private boolean isDeployedEnvironmentConcurrencyLimitPermitsDeployment(Deployment deployment) {
+        Integer concurrencyLimit = environmentDao.getEnvironment(deployment.getEnvironmentId()).getConcurrencyLimit();
+        if (concurrencyLimit != null && concurrencyLimit > 0) {
+            long runningDeploymentOnEnvironment = deploymentDao.getAllRunningDeployments()
+                    .stream()
+                    .filter(runningDeployment -> runningDeployment.getEnvironmentId() == deployment.getEnvironmentId())
```

Comment:
How many deployments do we have? If it's not a big number than we're good, otherwise consider filtering in the SQL (have a getAllRunningDeployments(int envId) added to the dao)

Reply:
Running deployment is a limited list, not worth adding SQL here IMO as in-mem is more readable